meteor/diffmaps.py

```diff
@@ -0,0 +1,195 @@
+import numpy as np
+import pandas as pd
+import reciprocalspaceship as rs
+
+from .settings import TV_MAP_SAMPLING
+from .utils import compute_map_from_coefficients
+from .validate import ScalarMaximizer, negentropy
+
+
+def _rs_dataseries_to_complex_array(
+    amplitudes: rs.DataSeries,
+    phases: rs.DataSeries,
+) -> np.ndarray:
+    pd.testing.assert_index_equal(
+        amplitudes.index, phases.index
+    )  # Ensure indices match
+    # Convert amplitudes and phases to complex representation
+    return amplitudes.to_numpy() * np.exp(1j * np.deg2rad(phases.to_numpy()))
+
+
+def _complex_array_to_rs_dataseries(
+    complex_array: np.ndarray,
+    index: pd.Index,
+) -> tuple[rs.DataSeries, rs.DataSeries]:
+    # Extract amplitude from the complex array (magnitude)
+    amplitudes = rs.DataSeries(np.abs(complex_array), index=index)
+    amplitudes = amplitudes.astype(rs.StructureFactorAmplitudeDtype())  # Convert dtype
+
+    # Extract phase from the complex array (angle, in degrees)
+    phases = rs.DataSeries(np.rad2deg(np.angle(complex_array)), index=index)
+    phases = phases.astype(rs.PhaseDtype())  # Convert dtype
+
+    return amplitudes, phases
+
+
+def compute_deltafofo(
```

Comment:
I like `compute_difference_map` better...

the map doesn't need to be an FoFo, you could use this for lots of things (FoFc, bla bla)

Reply:
fofo is in contrast with kweighted: what is difference map? what type?

Reply:
suggesting: compute_fofo_difference_map and compute_kweighted_difference_map